src/views/CardEditor.js

```diff
@@ -0,0 +1,67 @@
+import React, {useCallback, useContext} from 'react'
+import app from '../firebase';
+import {withRouter, Redirect} from 'react-router-dom'
+
+const EditorWindow = props => {
+    return (
+        <div className="editor">
+            {props.children}
+        </div>
+    )
+}
+
+const CardEditor = ({history}) => {
+    const handleSubmit = useCallback(
+        async event => {
+          event.preventDefault();
+          console.log(event.target.elements);
+          const { front, back, collection } = event.target.elements;
+          try {
+            await app.firestore().collection('cards').add({
+                front: front.value,
+                back: back.value,
+                collection: collection.value,
+                created: new Date(),
+                nextRecall: new Date(),
+                learningStage: 0,
+              }).catch(function(error) {
+                console.error('Error writing new message to Firebase Database', error);
```

Comment:
I think you can omit this `.catch` call. Your error will be caught below in `catch` block of `try \ catch` statement.